### PR TITLE
Add `environment` element at `board` and `component` level

### PR DESCRIPTION
**Pull request by silviooliva**
Commit related to issue #230.

### CODE REVIEW

**Comment by silviooliva** [comment]
I've introduced the `<environment>` element as child element of the `<board>` and the `<component>`.

Just a quick explanation hereafter concerning the modification to the `<component>` in the PACK.xsd schema.

The `<component>` element is defined as a `<xs:complexType>` and all its child elements are contained into the `<xs:all>` schema component.
Now, since the child elements in the `<xs:all>` can have only _maxOccur=1_, I cannot define the `<environment>` directly into the `<component>` in this simple way (that is with _maxOccurs="unbounded_):


```
<xs:element name="component" minOccurs="0" maxOccurs="unbounded">
  <xs:complexType>
    <xs:all>
      ... ... ... 
```
**`       <xs:element name="environment" type="EnvironmentType" minOccurs="0" maxOccurs="unbounded" />`**
```
      ... ... ... 
    </xs:all>
    <!-- component identity attributes -->
    <xs:attribute name="Cvendor" type="CvendorType" use="optional" />
    ... ... ...
  </xs:complexType>
</xs:element>
```

To overcome this issue, I had to define an **`<environments>`** element to be used as _container_ of the possible `<environment>` elements.
Here a snapshot of the xsd code:

```
<xs:element name="component" minOccurs="0" maxOccurs="unbounded">
  <xs:complexType>
    <xs:all>
      ... ... ...
```
``       <!-- list of environments (elements containing information that is specific for a development tool) -->``
**`       <xs:element name="environments" minOccurs="0" >`**
**`         <xs:complexType>`**
**`           <xs:sequence>`**
**`             <xs:element name="environment" type="EnvironmentType" minOccurs="1" maxOccurs="unbounded" />`**
**`           </xs:sequence>`**
**`         </xs:complexType>`**
**`       </xs:element>`**
```
      ... ... ... 
    </xs:all>
    <!-- component identity attributes -->
    <xs:attribute name="Cvendor" type="CvendorType" use="optional" />
    ... ... ...
  </xs:complexType>
</xs:element>

```
This solution, even if not aligned to how the `<environment>` is defined elsewhere in the standard (e.g. for the [device](https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/pdsc_family_pg.html#element_device) and the [part](https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/pdsc_parts_pg.html#element_part)), thanks to the `<environments>` _container_ allows a pack developer to declare unlimited `<environment>` elements.

File: doxygen/src/components_schema.txt [hunk 1]
```diff
@@ -593,20 +593,24 @@ A component describes a collection of files (source, header, configuration, libr
...
     <td>1</td>
   </tr>
-
   <tr>
     <td>\ref element_extensions "extensions"</td>
     <td>Grouping element for all extensions concerning this component.</td>
     <td>group</td>
     <td>1</td>
   </tr>
+  <tr>
```
Comment (jkrech) [inline doxygen/src/components_schema.txt:608]: We had previously agreed to add the "<extensions>" sub-element to the `<component>` element: https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/pdsc_components_pg.html#element_extensions.
```
Release Notes: 
1. June 2022: v1.7.8
   - added extension points description for components as suggested by STMicroelectronics
```
Do we need both? What has changed in between?

Comment (jkrech) [reply doxygen/src/components_schema.txt:608]: It would be good to see real examples for both elements I guess.

**Review by jkrech — APPROVED** [review]
Looks consistent to me.

**Pull request merged**